libs/designer/src/lib/ui/panel/templatePanel/templatePanel.tsx

```diff
@@ -1,23 +1,63 @@
 import { Panel, PanelType } from '@fluentui/react';
 import type { AppDispatch, RootState } from '../../../core/state/templates/store';
-import { useCallback } from 'react';
+import { useCallback, useMemo } from 'react';
 import { useDispatch, useSelector } from 'react-redux';
 import { closePanel } from '../../../core/state/templates/panelSlice';
 import { CreateWorkflowPanel } from './createWorkflowPanel/createWorkflowPanel';
 import { QuickViewPanel } from './quickViewPanel/quickViewPanel';
+import { type TemplatePanelTab, TemplatesPanelFooter, TemplatesPanelHeader } from '@microsoft/designer-ui';
+import { useCreateWorkflowPanelTabs } from './createWorkflowPanel/usePanelTabs';
+import { useQuickViewPanelTabs } from './quickViewPanel/usePanelTabs';
+import { clearTemplateDetails } from '../../../core/state/templates/templateSlice';
 
 export const TemplatePanel = ({ onCreateClick }: { onCreateClick: () => Promise<void> }) => {
   const dispatch = useDispatch<AppDispatch>();
-  const { isOpen, currentPanelView } = useSelector((state: RootState) => state.panel);
+  const { selectedTabId, isOpen, currentPanelView } = useSelector((state: RootState) => state.panel);
+  const templateTitle = useSelector((state: RootState) => state.template?.manifest?.title) ?? '';
+  const templateDescription = useSelector((state: RootState) => state.template?.manifest?.description) ?? '';
+  const dismissPanel = useCallback(() => {
+    dispatch(closePanel());
+    dispatch(clearTemplateDetails());
+  }, [dispatch]);
+  const createWorkflowPanelTabs = useCreateWorkflowPanelTabs(onCreateClick);
+  const quickViewPanelTabs = useQuickViewPanelTabs();
 
-  const dismissPanel = useCallback(() => dispatch(closePanel()), [dispatch]);
+  const currentPanelTabs: TemplatePanelTab[] = useMemo(
+    () => (currentPanelView === 'createWorkflow' ? createWorkflowPanelTabs : currentPanelView === 'quickView' ? quickViewPanelTabs : []),
+    [currentPanelView, createWorkflowPanelTabs, quickViewPanelTabs]
+  );
+
+  const selectedTabProps = selectedTabId ? currentPanelTabs?.find((tab) => tab.id === selectedTabId) : currentPanelTabs[0];
+
+  const onRenderHeaderContent = useCallback(
+    () => (
+      <TemplatesPanelHeader
+        title={currentPanelView === 'createWorkflow' ? 'Create a new workflow' : templateTitle}
+        description={currentPanelView === 'createWorkflow' ? templateDescription : 'By Microsoft'}
```

Comment:
The header content is different for quickview panel and createWorkflow panel... we have tags as well.. I can update this component when i merge my changes.